temporalio/worker/_workflow_instance.py

```diff
@@ -1596,6 +1651,48 @@ def _is_workflow_failure_exception(self, err: BaseException) -> bool:
             )
         )
 
+    def _warn_if_unfinished_handlers(self) -> None:
+        updates_warning = self._make_unfinished_update_handlers_warning()
+        if updates_warning:
+            warnings.warn(updates_warning, RuntimeWarning)
```

Comment:
I wonder if it'd be valuable making our own warning that extends `RuntimeWarning`. This could clean things up a tad, because you can call `warnings.warn(my_instance)` and it will use the class name as the category and the `str(my_instance)` as the message so you can move the string message building to `__str__()` of that instead of a constant at the top and concatenation down here (arguably limited value in a single-use message constant at the top compared to inline like other string literal messages for warnings and exceptions throughout). E.g.

```python
class UnfinishedSignalHandlerWarning(RuntimeWarning):
    def __init__(self, jobs: Sequence[temporalio.bridge.proto.workflow_activation.SignalWorkflow]) -> None:
        super().__init__()
        self.jobs = jobs

    def __str__(self) -> str:
        message = """ ... """.replace("\n", " ").strip()
        message += whatever_suffix_logic
        return message
```

Then here:

```python
    jobs = [
        job
        for job, defn in self._in_progress_updates.values()
        if defn.unfinished_handlers_policy
        == temporalio.workflow.UnfinishedHandlersPolicy.WARN_AND_ABANDON
    ]
    if jobs:
        warnings.warn(UnfinishedSignalHandlerWarning(jobs))
```

Also, I think we should turn these warnings into errors by default in our unit tests which gives even more justification to a specifically named/referencable one.